### PR TITLE
Change heavy automatic shotgun to actually use ammo belt

### PR DESCRIPTION
**Pull request by Rail-Runner**
Change heavy automatic shotgun to actually use ammo belt, and add 12 gauge ammo linkages and recipe for them. Not sure if forging is the most reasonable way to make ammo linkages, but I couldn't think of anything better.

Other changes:

1. Change heavy automatic shotgun material to steel, to keep it consistent with the gun that it was modified from.

2. Remove RELOAD_EJECT flag. Can't think of a reason why it was present on this gun.

3. Change reload time to 400 (was 800), like other belt-fed machineguns.

4. Remove sights gunmod slot, as this is a hip-fired weapon and therefore weapon user can't use normal scopes or sights while firing it.

5. Make gun use "modes" entry instead of "burst".

6. Change ammo belts names so that different ammo belts are more easily distinguishable from each other.

7. Add "shotgun_magazine" item class and change Saiga-12 to use it.

8. Add usable magazines for .223 and .308 conversions of Saiga-12.

9. Allow converting a 12 gauge shotgun to belt-feed with belt feed adapter. Currently works only for Saiga-12, as there are no other magazine-fed shotguns at the moment.

10. Add proper plural name to Saiga-12 to prevent "Saiga-12s" naming.

### CODE REVIEW

**Comment by Rivet-the-Zombie** [comment]
>Add proper plural name to Saiga-12 to prevent "Saiga-12s" naming.

I'm not sure which would be technically correct here, but I think that '3 Saiga-12s' sounds better than '3 Saiga-12'.

File: data/json/items/gun/shot.json [hunk 1]
```diff
@@ -1,5 +1,29 @@
 [
   {
+    "id": "abzats",
+    "copy-from": "shotgun_base",
```
Comment (mugling) [inline data/json/items/gun/shot.json]: This should probably actually `copy-from` the m2 browning as specified in the `description`

File: data/json/items/magazine/223.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
     "id": "belt223",
     "copy-from": "magazine_belt",
     "type": "MAGAZINE",
-    "name": "ammo belt",
+    "name": ".223 ammo belt",
```
Comment (mugling) [inline data/json/items/magazine/223.json]: Unrelated change

File: data/json/items/magazine/308.json [hunk 3]
```diff
@@ -3,7 +3,7 @@
     "id": "belt308",
     "copy-from": "magazine_belt",
     "type": "MAGAZINE",
-    "name": "ammo belt",
+    "name": ".308 ammo belt",
```
Comment (mugling) [inline data/json/items/magazine/308.json]: Also unrelated

File: data/json/items/magazine/40mm.json [hunk 4]
```diff
@@ -2,7 +2,7 @@
   {
     "id": "belt40mm",
     "type": "MAGAZINE",
-    "name": "grenade belt",
+    "name": "40mm grenade belt",
```
Comment (mugling) [inline data/json/items/magazine/40mm.json]: Similar (plus all that follow)

**Comment by mugling** [comment]
This is really complicated for what it's actually trying to achieve. I'm not sure creating a new ammo belt plus linkages just for that item is a good idea and extending the Saiga to use them is almost certainly a bad idea considering we already have the drum magazine (which is already far too good). An extra abstract class that is only used once isn't inheritance rather obfuscation.

File: data/json/items/gun/shot.json [hunk 5]
```diff
@@ -197,9 +221,10 @@
   },
   {
     "id": "saiga_12",
-    "copy-from": "shotgun_base",
+    "copy-from": "shotgun_magazine",
```
Comment (mugling) [inline data/json/items/gun/shot.json]: You only use this class once. It therefore doesn't reduce any redundancy and just obfuscates what's going on

File: data/json/items/gun/shot.json [hunk 6]
```diff
@@ -211,7 +236,11 @@
     "durability": 7,
     "barrel_length": 3,
     "magazine_well": 1,
-    "magazines": [ [ "shot", [ "saiga10mag", "saiga30mag" ] ] ],
+    "magazines": [
+      [ "shot", [ "saiga10mag", "saiga30mag" ] ],
+      [ "308", [ "falmag", "falbigmag" ] ],
```
Comment (mugling) [inline data/json/items/gun/shot.json]: Are these conversions actually possible?

**Comment by mugling** [comment]
Probably conflicts with #19430

**Comment by DangerNoodle** [comment]
Only possible conflict is due to the mod changing allowed magazines for the Saiga-12, due to not currently allowing conversion to .223 or .308. A change I am unsure about in realism terms, and in game terms no option to convert 12 gauge into a rifle chambering is implemented.

File: data/json/items/gun/shot.json (hunk 6 above)
Comment (Darkyhard) [inline data/json/items/gun/shot.json]: Saiga semi-automatic rifle variants which use .223 and .308 ammo exist. The only thing I'm not sure about is which magazine should be used for .308 conversion.

https://en.wikipedia.org/wiki/Saiga_semi-automatic_rifle

**Comment by Darkyhard** [comment]
> This is really complicated for what it's actually trying to achieve. I'm not sure creating a new ammo belt plus linkages just for that item is a good idea

Isn't this true for most guns with detachable magazines or ammo belts? Usually these magazines don't fit anything else (unless we're taking STANAG), so they are limited to be used exclusively with one gun, and one has to create a magazine too if converting a gun to use magazines. The only extra complications are addition of linkages and recipe for them (simply because there is no other possible source of linkages). Belt-fed conversion of Saiga-12 is added mostly for the purpose of not making 12 gauge ammo belt usable only by a single gun; it would also be possible to use it with other magazine-fed shotguns if we had more of them.

> It would be much simpler if the item was adapted to use Saiga drum magazines.

The original HMG is belt-fed, so that wouldn't make much sense.

> Nobody is going to craft a bunch of obscure linkages using a high level recipe

The recipe for linkages uses fabrication skill and has a difficulty of 6, which is the same skill that is used to make heavy automatic shotgun. In fact, the shotgun itself also requires mechanics 6 and gun 4 skill on top of that to craft, so linkages recipe is very likely to become available even before the gun itself, or at worst it will become available right after the gun becomes available. The only possible problem is that linkages recipe requires forging tools, but if character has fabrication 6, it means that it likely already has access to such tools. I'd be happy to change it to use welder or a firearm/gunsmith repair kit instead of a forge, but I'm not sure if this is possible.

**Comment by DangerNoodle** [comment]
The linkages might be more useful if normal or at least autoloading shotguns supported the option to convert to belt feed.

**Comment by Rivet-the-Zombie** [comment]
>Nobody is going to craft a bunch of obscure linkages using a high level recipe when they can just use a regular Saiga with a drum magazine or the crafted automatic item.

Part of the point of craftable guns is that you don't have to hope to find magazines; you can make your own.

**Comment by DangerNoodle** [comment]
That also works. But I am wondering what the balance implications would be if this item was useful for other weapons. What would be a reasonable ammo belt size if that were the case?

**Comment by Rivet-the-Zombie** [comment]
>What would be a reasonable ammo belt size if that were the case?

I'd go with twenty.

**Comment by DangerNoodle** [comment]
That sounds reasonable, and an even tenth of the most common number used by the other ammo belts. I was thinking that a number higher than the larger Saiga magazine, while still being less than 100, would work too.

But then again, this item is presumed to be improvised only, and would have to be balanced against other autoloading shotguns if allowed for more firearms, rather than solely against the automatic shotgun and/or the Saiga-12.

File: data/json/items/generic/ammolink.json [hunk 7]
```diff
@@ -54,5 +54,15 @@
       "type": "ammobelt",
       "belt": "belt50"
     }
+  },
+  {
+    "id": "ammolinkshot",
+    "copy-from": "ammolink",
+    "type": "GENERIC",
+    "name": "12 gauge ammo belt linkage",
```
Comment (mugling) [inline data/json/items/generic/ammolink.json]: Minor: Too lengthy, shot belt linkage would be reasonable (or 12g belt linkage)

File: data/json/items/gun/shot.json [hunk 8]
```diff
@@ -1,5 +1,32 @@
...
+    "ammo": "shot",
+    "skill": "shotgun",
+    "modes": [
+      [ "DEFAULT", "semi-auto", 1 ],
+      [ "AUTO", "auto", 5 ]
+    ],
+    "valid_mod_locations": [ [ "accessories", 4 ], [ "mechanism", 4 ], [ "rail", 1 ] ],
+    "magazines": [ [ "shot", [ "beltshot" ] ] ],
+    "relative": {
+      "weight": -16000,
```
Comment (mugling) [inline data/json/items/gun/shot.json]: It's supposed to be hugely impractical! If you want a rambo gun it should be rambo's only gun and require a character like him to make use of it. You shouldn't drop 16kg of weight here

File: data/json/items/gun/shot.json [hunk 9]
```diff
@@ -1,5 +1,32 @@
...
+      "volume": -2,
+      "price": -1050000,
```
Comment (mugling) [inline data/json/items/gun/shot.json]: Price is pretty much irrelevant - I wouldn't worry about tweaking it here

File: data/json/items/gun/shot.json [hunk 10]
```diff
@@ -1,5 +1,32 @@
...
+      "to_hit": -2,
```
Comment (mugling) [inline data/json/items/gun/shot.json]: Don't adjust `to_hit`

File: data/json/items/gun/shot.json [hunk 11]
```diff
@@ -1,5 +1,32 @@
...
+      "dispersion": -50,
+      "durability": -2,
+      "loudness": 25,
```
Comment (mugling) [inline data/json/items/gun/shot.json]: `loudness` doesn't work at the moment, consider dropping

File: data/json/items/gun/shot.json (hunk 6 above)
Comment (mugling) [inline data/json/items/gun/shot.json]: No ammo conversion kit exists for this (and doesn't need adding). As such those entries would never be used

File: data/json/items/gunmod/magazine.json [hunk 12]
```diff
@@ -12,13 +12,14 @@
     "symbol": ":",
     "color": "green",
     "location": "magazine",
-    "mod_targets": [ "rifle" ],
-    "acceptable_ammo": [ "223", "308", "50" ],
+    "mod_targets": [ "rifle", "shotgun" ],
+    "acceptable_ammo": [ "223", "308", "50", "shot" ],
```
Comment (mugling) [inline data/json/items/gunmod/magazine.json]: Please drop `shot`. Craftable guns specifically intended for belts is acceptable, converting factory firearms to snowflake-tier weapons isn't

File: data/json/items/magazine/shot.json [hunk 13]
```diff
@@ -1,5 +1,15 @@
 [
   {
+    "id": "beltshot",
+    "copy-from": "magazine_belt",
+    "type": "MAGAZINE",
+    "name": "12 gauge ammo belt",
+    "ammo_type": "shot",
+    "capacity": 100,
+    "count": 100,
+    "linkage": "ammolinkshot"
```
Comment (mugling) [inline data/json/items/magazine/shot.json]: Not required here but I am starting to wonder if we actually need all the different sizes of linkages versus a generic type. Ping @Rivet-the-Zombie (reality crowd) and @Coolthulhu (gameplay crowd)

**Comment by Darkyhard** [comment]
Done.

**Comment by mugling** [comment]
> Done.

Still have two outstanding review points

File: data/json/items/magazine/shot.json (hunk 13 above)
Comment (Rivet-the-Zombie) [inline data/json/items/magazine/shot.json]: RL-wise, linkages are particular to a single caliber of bullet- (they need to fit snugly around the casing so they stay properly linked.

For something like our abzats shotgun, I'd go with an old-fashioned cloth ammo belt due to its ease of construction and reusability compared to disintegrating linkages.

File: data/json/items/gun/shot.json (hunk 6 above)
Comment (BorkBorkGoesTheCode) [inline data/json/items/gun/shot.json]: Currently, the 223 and 308 conversion kits support converting a shotgun into a rifle. There appear to be some websites advertising Saiga-12 conversion kits; however, I haven't checked them out.

Comment (mugling) [inline data/json/items/gun/shot.json]: My error, they do exist - but they shouldn't. Again realism (if that's your thing due to no rifling) but also gameplay (everything doesn't need converting to everything else)

File: data/json/items/magazine/shot.json (hunk 13 above)
Comment (Darkyhard) [inline data/json/items/magazine/shot.json]: How exactly are they made? Would a sewing kit and some rags work?

Comment (mugling) [inline data/json/items/magazine/shot.json]: Sounds reasonable

Comment (mugling) [inline data/json/items/magazine/shot.json]: Or better still a lot of thread

**Comment by mugling** [comment]
Stalled (cloth belt and ammo conversions)

**Comment by kevingranade** [comment]
Feel free to reopen once the issues are addressed.
